From badd991784d27dc6cb07b0cc034bf31a0f07de93 Mon Sep 17 00:00:00 2001
From: Anthoine Bourgeois <anthoine.bourgeois@vates.tech>
Date: Thu, 27 Feb 2025 17:41:36 +0100
Subject: [PATCH] blkif: add a comment on memory barrier usage

The memory barrier is mandatory to see the queued requests from blkif
frontend.

No functional change expected.

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@vates.tech>
---
 drivers/td-ctx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/td-ctx.c b/drivers/td-ctx.c
index a1da30f6..8b75dc48 100644
--- a/drivers/td-ctx.c
+++ b/drivers/td-ctx.c
@@ -220,7 +220,7 @@ __xenio_blkif_get_requests(struct td_xenblkif * const blkif,
     ring = &blkif->rings.common;
 
     rp = ring->sring->req_prod;
-    xen_rmb(); /* TODO why? */
+    xen_rmb(); /* Ensure we see queued requests up to 'rp'. */
 
     for (rc = ring->req_cons, n = 0, barrier = false;
 			rc != rp && n < count && !barrier;
